data_model/derive/src/model.rs

```diff
@@ -1,8 +1,49 @@
 use proc_macro2::TokenStream;
-use quote::quote;
+use proc_macro_error::abort;
+use quote::{quote, ToTokens};
 use syn::{parse_quote, Attribute};
 
-pub fn process_item(mut input: syn::DeriveInput) -> TokenStream {
+pub fn impl_model(input: &syn::ItemMod) -> TokenStream {
+    let syn::ItemMod {
+        attrs,
+        vis,
+        mod_token,
+        ident,
+        content,
+        semi,
+        ..
+    } = input;
+
+    let syn::Visibility::Public(vis_public) = vis else {
+        abort!(input, "The `model` attribute can only be used on public modules");
+    };
+    if ident != "model" {
+        abort!(
+            input,
+            "The `model` attribute can only be used on the `model` module"
+        );
+    }
+
+    let items_code = content.as_ref().map_or_else(Vec::new, |(_, items)| {
+        items.iter().cloned().map(process_item).collect()
+    });
+
+    quote! {
+        #(#attrs)*
+        #[doc = "Module with exported structs and enums forming *model* of the parent module"]
```

Comment:
```suggestion
        #[allow(missing_docs)]
```

I would use `allow(missing_docs)` here. We use it already in similar situations